packages/debug/src/browser/view/debug-session-widget.ts

```diff
@@ -82,17 +88,18 @@ export class DebugSessionWidget extends BaseWidget implements StatefulWidget, Ap
         this.title.label = this.model.label;
         this.title.caption = this.model.label;
         this.title.closable = true;
-        this.title.iconClass = 'debug-tab-icon';
+        this.title.iconClass = DebugWidget.ICON_CLASS;
         this.addClass('theia-session-container');
 
         this.viewContainer = this.viewContainerFactory({
             id: 'debug:view-container:' + this.model.id
         });
-        this.viewContainer.addWidget(this.threads, { weight: 30 });
-        this.viewContainer.addWidget(this.stackFrames, { weight: 20 });
-        this.viewContainer.addWidget(this.variables, { weight: 10 });
-        this.viewContainer.addWidget(this.watch, { weight: 10 });
-        this.viewContainer.addWidget(this.breakpoints, { weight: 10 });
+        this.viewContainer.setTitleOptions(DEBUG_VIEW_CONTAINER_TITLE_OPTIONS);
+        this.viewContainer.addWidget(this.threads, { weight: 30, disableDraggingToOtherContainers: true });
+        this.viewContainer.addWidget(this.stackFrames, { weight: 20, disableDraggingToOtherContainers: true });
+        this.viewContainer.addWidget(this.variables, { weight: 10, disableDraggingToOtherContainers: true });
+        this.viewContainer.addWidget(this.watch, { weight: 10, disableDraggingToOtherContainers: true });
+        this.viewContainer.addWidget(this.breakpoints, { weight: 10, disableDraggingToOtherContainers: true });
```

Comment:
@colin-grant-work I disabled dragging `debug` widgets because there's a problem with restoring them to the last container they were dragged to after refresh, Do you think it still worth to enable dragging theme although they won't get restored after refresh (but return to the `debug` container?
About the icon - it's very strange because I do show this icon when trying to drag such views...

Reply:
I think this can also be omitted for now if it's going to be difficult to implement - the general functionality is present. Is there a reason the approach taken to restoring the state of other view containers that contain migratory view container parts doesn't apply here?

Reply:
Only for the `debug` widgets [this property] (https://github.com/EstherPerelman/theia/blob/b6b0f1be2375fd454fefbbbf28895bdffe06fc69/packages/core/src/browser/view-container.ts#L499) is correctly save on storing but is coming `undefined` on [restore](https://github.com/EstherPerelman/theia/blob/b6b0f1be2375fd454fefbbbf28895bdffe06fc69/packages/core/src/browser/view-container.ts#L519) which make it impossible to create it in the right place, I can guess it has something with the different behavior of the `debug` session widget.

Reply:
So it looks like what's happening here is this. When the `ShellLayoutRestorer` is storing the layout, if it encounters a `Widget` it tries to convert that widget to a description:

https://github.com/EstherPerelman/theia/blob/84bca9b36ac6a2029a57d9406d8c13f695aa0397/packages/core/src/browser/shell/shell-layout-restorer.ts#L205-L221

To do that, it asks the `WidgetManager` to get the description, and that only works _if the widget was created through the `WidgetManager`_

https://github.com/EstherPerelman/theia/blob/84bca9b36ac6a2029a57d9406d8c13f695aa0397/packages/core/src/browser/widget-manager.ts#L237-L250

But, the `DebugSessionWidget` parts are created without going through the `WidgetManager`

https://github.com/EstherPerelman/theia/blob/84bca9b36ac6a2029a57d9406d8c13f695aa0397/packages/debug/src/browser/view/debug-session-widget.ts#L41-L45

So they always wind up with no description. I think that that is a bad design on the part of the `debug` package, because it means that it expects to create a whole set of widgets for each session and then hide and reveal them depending on which session is selected. Instead, I think it would be better to have the same widgets able to hook up to whatever session is selected, but that's a major refactor of the `debug` package, so I don't think it should be undertaken as part of this PR. It seems like disabling drag-drop there is probably best for now, then.

Reply:
Yea, all the issues with the `debug` container are related to the missing creation by `widgetManager`, During the tries to solve this I thought to add it manually to the widgets list (not by factory creation), but it seemed to be incorrect according to the `widgetManager` behavior, 
Actually I didn't catch that resorting `debug` widgets issue is also related to this, thank you for the clear explanation!

Reply:
I'm also in favor of disabling dnd for debug widgets and proceeding with this PR 👍